### PR TITLE
support the new stm32L5xx serie 

### PR DESCRIPTION
**Pull request by FRASTM**
add the new stm32L5xx series of Ultra-low-power MCUs
Ultra-low-power with FPU Arm Cortex-M33 with Trust Zone, MCU 110 MHz with 512 kbytes of Flash .
**[https://www.st.com/en/microcontrollers-microprocessors/stm32l5-series.html](url)**

Note that **TZ mode is not supported** in this PR (TZEN = 0)

From [STM32CubeL5 release 1.3.1](https://github.com/STMicroelectronics/STM32CubeL5)

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: system/STM32L5xx/stm32l5xx_hal_conf_default.h [hunk 1]
```diff
@@ -0,0 +1,497 @@
+/**
+  ******************************************************************************
+  * @file    stm32l5xx_hal_conf_default.h
+  * @brief   HAL default configuration file.
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2017 STMicroelectronics.
```
Comment (fpistm) [inline system/STM32L5xx/stm32l5xx_hal_conf_default.h:8]: This file is not correct as it does not match to the stm32l5xx_hal_conf_template.h

**Review by fpistm — CHANGES_REQUESTED** [review]
[L5] sort version.md files commit have to be split and merged with the HAL and CMSIS.

**Comment by FRASTM** [comment]
[L5] sort version.md files commit have to be split and merged with the HAL and CMSIS
respectively (not sorted by the script, though)

File: system/STM32L5xx/stm32l5xx_hal_conf_default.h (hunk 1 above)
Comment (FRASTM) [reply system/STM32L5xx/stm32l5xx_hal_conf_default.h:8]: The `system/STM32L5xx/stm32l5xx_hal_conf_default.h` 
is built like the 
`system/STM32L4xx/stm32l4xx_hal_conf_default.h`

**Comment by FRASTM** [comment]
change the `cores/arduino/stm32/stm32_def_build.h`
after 7f7bf8af28716e316b708168c4091e8c79116799

File: system/STM32L5xx/stm32l5xx_hal_conf_default.h [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+/* ########################### System Configuration ######################### */
+/**
+  * @brief This is the HAL system configuration section
+  */
+#if !defined (VDD_VALUE)
+#define VDD_VALUE          3300UL /*!< Value of VDD in mv */
+#endif
+#if !defined (TICK_INT_PRIORITY)
+#define TICK_INT_PRIORITY  ((1UL<<__NVIC_PRIO_BITS) - 1UL)  /*!< tick interrupt priority (lowest by default) */
```
Comment (fpistm) [inline system/STM32L5xx/stm32l5xx_hal_conf_default.h:177]:
```suggestion
#define  TICK_INT_PRIORITY            (0x00UL) /*!< tick interrupt priority */
```

File: system/STM32L5xx/system_stm32l5xx.c [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  #define HSE_VALUE    16000000U /*!< Value of the External oscillator in Hz */
+#endif /* HSE_VALUE */
+
+#if !defined  (MSI_VALUE)
+  #define MSI_VALUE    4000000U  /*!< Value of the Internal oscillator in Hz*/
+#endif /* MSI_VALUE */
+
+#if !defined  (HSI_VALUE)
+  #define HSI_VALUE    16000000U /*!< Value of the Internal oscillator in Hz*/
+#endif /* HSI_VALUE */
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply system/STM32L5xx/system_stm32l5xx.c]: done

File: system/STM32L5xx/stm32l5xx_hal_conf_default.h (hunk 2 above)
Comment (FRASTM) [reply system/STM32L5xx/stm32l5xx_hal_conf_default.h:177]: some changes to come in the libraries/SrcWrapper and cores/arduino/stm32/timer.h  to support the STM32L5xx serie

**Comment by FRASTM** [comment]
setting the Vddio2 Independent I/Os supply is required to have LPUART1 Tx/Rx pins on PG7/PG8
--> Serial.printing OK  (@ 9600 on LSE or 115200)

**Comment by FRASTM** [comment]
rebase

**Review by fpistm — APPROVED** [review]
LGTM

**Pull request merged**